### PR TITLE
mdn folder to md html prep

### PR DESCRIPTION
**Pull request by Rumyra**
HTML updates to the /mdn folder before markdown conversion

### CODE REVIEW

**Comment by Rumyra** [comment]
There's a couple of comments from the last pr that I'll take a look at before this is ready

File: files/en-us/mdn/contribute/howto/create_an_interactive_exercise_to_help_learning_the_web/index.html [hunk 1]
```diff
@@ -25,25 +25,20 @@ <h2 id="MDN_live_samples">MDN live samples</h2>
...
 
 <p>Let's see a simple example:</p>
 
-<div class="moreinfo">
+<div>
 <p>Click on the following square to randomly change its color:</p>
 
-<div class="hidden">
-<h4 id="hidden_code_example">hidden code example</h4>
+<div id="hidden_code_example">
```
Comment (wbamberg) [inline files/en-us/mdn/contribute/howto/create_an_interactive_exercise_to_help_learning_the_web/index.html:35]: This technique although technically still possible in Markdown should be deprecated IMO. But that's out of scope for this conversion change.

File: files/en-us/mdn/contribute/howto/write_an_api_reference/index.html [hunk 2]
```diff
@@ -10,9 +10,7 @@
 ---
 <div>{{MDNSidebar}}</div>
 
-<div class="summary">
-<p><span class="seoSummary">This guide takes you through all you need to know to write an API reference on MDN.</span></p>
-</div>
+<p>This guide takes you through all you need to know to write an API reference on MDN.</span></p>
```
Comment (wbamberg) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/index.html:13]:
```suggestion
<p>This guide takes you through all you need to know to write an API reference on MDN.</p>
```

File: files/en-us/mdn/contribute/processes/matching_features_to_browser_version/index.html [hunk 3]
```diff
@@ -381,6 +381,6 @@ <h3 id="Useful_links">Useful links</h3>
  </li>
 </ul>
 
-<div class="notecard warning">
```
Comment (wbamberg) [inline files/en-us/mdn/contribute/processes/matching_features_to_browser_version/index.html:384]: FYI we don't have to remove `notecard`.

File: files/en-us/mdn/guidelines/css_style_guide/index.html [hunk 4]
```diff
@@ -11,12 +11,12 @@
 ---
 <p>{{MDNSidebar}}</p>
 
-<div class="notecard warning">
+<div class="warning">
   <p><strong>Warning:</strong> We're currently in the process of converting all MDN content from HTML into Markdown. Currently you can write pages in HTML or in Markdown, but we don't want to mix formats within a section.</p>
-  <p>At the moment only the JavaScript documentation is in Markdown (pages under <a href="/en-US/docs/Web/JavaScript">https://developer.mozilla.org/en-US/docs/Web/JavaScript</a>). So if you're writing content in that part of the site, this page is <em>not</em> applicable, and you should instead follow the guidelines in the <a href="/en-US/docs/MDN/Contribute/Markdown_in_MDN">Markdown in MDN</a> guide.</p>
+  <p>At the moment CSS, JavaScript and Web API documentation is in Markdown. So if you're writing content in that part of the site, this page is <em>not</em> applicable, and you should instead follow the guidelines in the <a href="/en-US/docs/MDN/Contribute/Markdown_in_MDN">Markdown in MDN</a> guide.</p>
```
Comment (wbamberg) [inline files/en-us/mdn/guidelines/css_style_guide/index.html:16]:
```suggestion
  <p>At the moment the CSS, JavaScript, and Web API documentation is in Markdown. So if you're writing content in that part of the site, this page is <em>not</em> applicable, and you should instead follow the guidelines in the <a href="/en-US/docs/MDN/Contribute/Markdown_in_MDN">Markdown in MDN</a> guide.</p>
```

File: files/en-us/mdn/contribute/markdown_in_mdn/index.html [hunk 5]
```diff
@@ -6,13 +6,11 @@
...
-<div class="notecard warning">
+<div class="warning">
 
-<h4>Warning</h4>
+  <p><strong>Warning:</strong> We're currently in the process of converting all MDN content from HTML into Markdown. Currently you can write pages in HTML or in Markdown, but we don't want to mix formats within a section.</p>
 
-<p>We're currently in the process of converting all MDN content from HTML into Markdown. Currently you can write pages in HTML or in Markdown, but we don't want to mix formats within a section.</p>
-
-<p>At the moment only the JavaScript documentation is in Markdown (pages under <a href="/en-US/docs/Web/JavaScript">https://developer.mozilla.org/en-US/docs/Web/JavaScript</a>), so this page only applies to that part of the site. For other parts of the site, write documentation in HTML.</p>
+  <p>At the moment JavaScript, CSS and Web API documentation are in Markdown, so this page applies to those parts of the site. For other parts of the site, write documentation in HTML.</p>
```
Comment (wbamberg) [inline files/en-us/mdn/contribute/markdown_in_mdn/index.html:13]:
```suggestion
  <p>At the moment the JavaScript, CSS, and Web API documentation are in Markdown, so this page applies to those parts of the site. For other parts of the site, write documentation in HTML.</p>
```

File: files/en-us/mdn/structures/syntax_sections/index.html [hunk 6]
```diff
@@ -254,7 +251,7 @@ <h4 id="Syntax_section">Syntax section</h4>
 
 <p>The "Syntax" section shows what a header's syntax will look like, using a syntax block styled using the "Syntax Box" style, including formal syntax to show exactly what directives can be included in the value, in what order, etc. For example, the {{HTTPHeader("If-None-Match")}} header's syntax block looks like this:</p>
 
-<pre class="brush: http">If-None-Match: &lt;etag_value&gt;
+<pre class="brush: html">If-None-Match: &lt;etag_value&gt;
```
Comment (wbamberg) [inline files/en-us/mdn/structures/syntax_sections/index.html:254]:
```suggestion
<pre class="brush: plain">If-None-Match: &lt;etag_value&gt;
```

File: files/en-us/mdn/structures/syntax_sections/index.html [hunk 7]
```diff
@@ -268,7 +265,7 @@ <h3 id="HTTP_request_methods">HTTP request methods</h3>
 
 <p>Request method syntax is really simple, just containing a syntax block styled using the "Syntax Box" style that shows how the method syntax is structured. The syntax for the <a href="/en-US/docs/Web/HTTP/Methods/GET">GET method</a> looks like this:</p>
 
-<pre class="brush: http">GET /index.html</pre>
+<pre class="brush: html">GET /index.html</pre>
```
Comment (wbamberg) [inline files/en-us/mdn/structures/syntax_sections/index.html:268]:
```suggestion
<pre class="brush: plain">GET /index.html</pre>
```

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks for this PR! I've checked it and pushed a few little changes, but what's there looks fine. There are a few more things we could convert here with a bit more cleanup. I ran the converter over this and it listed these unconverted elements:

```
- kbd (15)
- td (14)
- tr (14)
- dl (4)
- pre.brush:.none (4)
- table.standard-table (3)
- div.example-good (3)
- table.properties (2)
- div.example-bad (2)
- div.notecard.warning (2)
- div#audio (1)
- figure (1)
- div#hidden_code_example (1)
- sup (1)
- div#Hidden_code (1)
- div.note (1)
- div.notecard.note (1)
- pre.brush:.wasm (1)
- pre.brush:.rust (1)
- table.fullwidth-table (1)
- div.hidden (1)
- ol (1)
- h4#How_to_write_in_text (1)
```

Of these I reckon we could fix the following.

## dl

These are a bit tricky.

The failures are in:
https://developer.mozilla.org/en-US/docs/MDN/Contribute/Howto/Add_or_update_browser_compatibility_data
https://developer.mozilla.org/en-US/docs/MDN/Contribute/Howto/Convert_code_samples_to_be_live
https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide
https://developer.mozilla.org/en-US/docs/MDN/Tools/KumaScript

The first two are because the converter doesn't like `<dd>` elements that don't start with text. This is a bug in the converter, but I have never seen a case of this which wasn't a content problem. In this case, these `<dl>` elements are surely not semantically `<dl>` elements: the pages should replace the `<dl>` with a series of `<h2>` elements for each question, like:

```
## Where does it need to be done?

Any article that has a static code example (just blocks of HTML, CSS, JavaScript) that you think would benefit from being shown running live.

## What do you need to know to do the task?

- Understanding of HTML, CSS and/or JavaScript, depending on the code sample.
- Ability to use KumaScript macros within MDN articles.
```

The second two don't convert because they contain consecutive `<dd>` elements, which we don't support. Although consecutive `<dd>` elements are allowed in HTML, again I've never seen a case on MDN which had them that wasn't an error, and that's true here too (each case should be a single `<dd>` containing two paragraphs).

## pre.brush:.none

These are all in https://developer.mozilla.org/en-US/docs/MDN/Tools/KumaScript , and would convert if we change it to `brush: plain`.

## div.example-good, div.example-good

These are all in https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide and should just be removed - the classes don't work properly on anything except code blocks.

## div.notecard.note, div.notecard.warning

These are in the https://developer.mozilla.org/en-US/docs/MDN/Kitchensink, and should be updated to use the proper style.

## div.note

This is in https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide, line 396, and just looks like a misplaced colon.

## figure

In similar cases I've just used an `` and had a paragraph above or below for a caption.

## pre.brush:.wasm, pre.brush:.rust

These are interesting. The converter doesn't know about these languages, although they are supported in Yari. The easiest thing to do is fix up the Markdown manually after conversion.

## ol

This is in https://developer.mozilla.org/en-US/docs/MDN/Structures/Macros/Commonly-used_macros, and has broken markup (a `<ul>` directly under the `<ol>`).

*******

(sorry to be picky. I know we will probably retire a lot of this content, but in the end I spent long enough looking at it that I thought it would be worth fixing these last bits.

**Comment by wbamberg** [comment]
Also, in https://developer.mozilla.org/en-US/docs/MDN/Tools/KumaScript, 'How to write "{{" in text' is a nightmare. I think the only way to get it to work in Markdown is to keep the whole thing in HTML. I have no idea why.

File: files/en-us/mdn/contribute/howto/create_an_interactive_exercise_to_help_learning_the_web/index.html (hunk 1 above)
Comment (Rumyra) [reply files/en-us/mdn/contribute/howto/create_an_interactive_exercise_to_help_learning_the_web/index.html:35]: OK I'll try and remember that when I'm moving docs over - maybe if we don't say it's a thing we'll automatically reduce the use of it

File: files/en-us/mdn/contribute/processes/matching_features_to_browser_version/index.html (hunk 3 above)
Comment (Rumyra) [reply files/en-us/mdn/contribute/processes/matching_features_to_browser_version/index.html:384]: :) I'll remember that for the future 👍

**Comment by Rumyra** [comment]
OK that should be the tidying up - some notes:

* Some of the `<dl>`s are diffficult to tidy up
* I wasn't going to convert kitchensink for now until we can rewrite it once everything is in markdown - as it's used for testing so I thought just in case... I have made the changes tho

Thanks so much @wbamberg

**Review by wbamberg — APPROVED** [review]
Thanks for the updates Ruth, this looks great!

**Comment by Rumyra** [comment]
Thanks @wbamberg - I'm going to do conversion next week as I'm now off until Monday 👍

**Pull request merged**